sycl/tools/CMakeLists.txt

```diff
@@ -20,8 +20,9 @@ target_link_libraries(get_device_count_by_type
     LevelZeroLoader::Headers
     OpenCL-ICD
     ${LEVEL_ZERO_LIBRARY}
-    $<$<BOOL:${SYCL_BUILD_PI_CUDA}>:cudadrv>
-    $<$<BOOL:${SYCL_BUILD_PI_ROCM}>:rocmdrv>
+    $<$<OR:$<BOOL:${SYCL_BUILD_PI_CUDA}>,$<AND:$<BOOL:${SYCL_BUILD_PI_ROCM}>,$<STREQUAL:${SYCL_BUILD_PI_ROCM_PLATFORM},"NVIDIA">>>:cudadrv>
+    $<$<AND:$<BOOL:${SYCL_BUILD_PI_ROCM}>,$<STREQUAL:${SYCL_BUILD_PI_ROCM_PLATFORM},"AMD">>:rocmdrv>
+    $<$<AND:$<BOOL:${SYCL_BUILD_PI_ROCM}>,$<STREQUAL:${SYCL_BUILD_PI_ROCM_PLATFORM},"NVIDIA">>:cudart>
```

Comment:
This is because HIP uses a lot of entry points from `cudart` rather than `cudadrv`, for example `hipHostGetDevicePointer` is turned into `cudaHostGetDevicePointer` by HIP, which is defined in `cudart`, but the CUDA plugin just uses `cuMemHostGetDevicePointer` which is defined in `cudadrv`. And this is the case for quite a few entry points.

So it's just that this is what HIP uses, not that there's any specific functionality we need from it.